### ISSUE TITLE
[DNN] Investigate image classification python example

### ISSUE THREAD

**Issue by huningxin** [huningxin/opencv#313]
https://github.com/opencv/opencv/blob/master/samples/dnn/classification.py

**Comment by huningxin** [comment]
@akineeic , we need to know:
1. how to get the models?
2. what models are supported?
3. what are the functions (input/output, UI etc.,) of the example?

These information are helpful for you define the js example. Please update your progress in this issue. Thanks!

**Comment by huningxin** [comment]
The js tutorial should showcase:
1. fetch model from network and save to emscripten fs (I/O)
2. load model into memory
3. blob from image and pre-processing
4. run model
5. post-processing and classification result
6. get performance data

**Comment by huningxin** [comment]
@akineeic , please create a table that lists all the image classification models the python example supports, including the URL, pre-processing, post-processing and results.

**Comment by akineeic** [comment]
I have tested the models listed in opencv [wiki](https://github.com/opencv/opencv/wiki/Deep-Learning-in-OpenCV). The processing information mainly includes mean, std, softmax, RGB or BGR and the labels txt file. The models predict correctly yet are as follows: 

- caffe

Caffe Model | mean | std | softmax | rgb | labels txt
:------------: | :-------------: | :-------------: | :-------------: | :-------------: | :-------------: 
alexnet | 104, 117, 123 | 1 | no | true  | labels1000
densenet | 127.5, 127.5, 127.5 | 0.007843 | yes | true  | labels1000
googlenet | 104, 117, 123 | 1 | no | true  | labels1000
squeezenet | 104, 117, 123 | 1 | no | true  | labels1000
VGG | 104, 117, 123 | 1 | no | true  | labels1000

- tensorflow

Tensorflow Model | mean | std | softmax | rgb | labels txt
:------------: | :-------------: | :-------------: | :-------------: | :-------------: | :-------------: 
inception | 123, 117, 104 | 1 | no | false  | labels1001
mobilenet | 127.5, 127.5, 127.5 | 0.007843 | no | false  | labels1002

- onnx model

Onnx Model | mean | std | softmax | rgb  | labels txt
:------------: | :-------------: | :-------------: | :-------------: | :-------------: | :-------------: 
alexnet | 104, 117, 123 | 1 | no | true  | labels1000
caffenet | 104, 117, 123 | 1 | no | true  | labels1000
densenet | 127.5, 127.5, 127.5 | 0.007843 | yes | false  | labels1000
googlenet | 104, 117, 123 | 1 | no | true  | labels1000
inception | 104, 117, 123 | 1 | no | true  | labels1000
resnet | 127.5, 127.5, 127.5 | 0.007843 | yes | false  | labels1000
shuffenet | 127.5, 127.5, 127.5 | 0.007843 | no | true  | labels1000
squeezenet | 104, 117, 123 | 1 | no | true  | labels1000
VGG | 123, 117, 104| 1 | yes | false  | labels1000
zfnet512 | 104, 117, 123 | 1 | no | true | labels1000

**Comment by huningxin** [comment]
please update the status of darknet, then we can close it.

**Comment by akineeic** [comment]
Most models mentioned in the [wiki](https://github.com/opencv/opencv/wiki/Deep-Learning-in-OpenCV) except darknet have been tested and update in the above table. The inference result of darknet is incorrect yet that may caused by the parameters.
